### DISCUSSION TITLE
Pictures of my setup with HM-800 + Pylontech + Victron + OpenDTU

### DISCUSSION THREAD

**Discussion by berni2288** [hoylabs/OpenDTU-OnBattery#128] [Ideas]
My setup:

- Victron MPPT 100/20 48V
- Pylontech US3000C 48V
- Hoymiles HM-800
- ESP32 NodeMCU on soldered PCB board
- Waveshare SN65HVD230 CAN Board

![Powerlimiter PCB Design](https://user-images.githubusercontent.com/1029328/223213432-2c663048-69a9-4830-a7b4-0f00be8939b4.jpg)

![PXL_20230306_193616197](https://user-images.githubusercontent.com/1029328/223213806-bb5cc2ea-7703-4cbe-abd2-e2bf7be75fae.jpg)

![PXL_20230306_185857963](https://user-images.githubusercontent.com/1029328/223213907-0be0cde1-fc1c-4770-a87c-7c6ee70b0049.jpg)

![PXL_20230306_185823868](https://user-images.githubusercontent.com/1029328/223214136-0daed02e-40ea-4839-ac53-d3d30e801ed6.jpg)

![PXL_20230306_185752761](https://user-images.githubusercontent.com/1029328/223214177-542ee549-0367-4caa-bd87-b678a0bfed88.jpg)

**Comment by FSys77** [comment]
Frage zum 2. Bild: Was ist das für ein Kasten in den du das verbaut hast? Sowas will ich mir auch zulegen.

**Reply by berni2288** [reply]
https://www.amazon.de/gp/product/B00R3HS41U/ref=ppx_yo_dt_b_asin_title_o04_s00?ie=UTF8&psc=1

**Comment by Head** [comment]
Hi @berni2288, ich klemm mich hier auch nochmal rein, auch wenn ich schon hier https://github.com/helgeerbe/OpenDTU-OnBattery/discussions/111#discussioncomment-5258195 am fragen bin.
Du hast es ja scheinbar so laufen, oder war dies nur ein Test-Betrieb? Fehlt hier nicht die Kommunikation zwischen Pylontech und SmartSolar?
Laut Matthias Kommentar in https://community.victronenergy.com/questions/55688/multiplus-ii-mit-pylontech-ohne-gx-moglich.html kann dein Aufbau gar nicht funktionieren, oder ist riskant, weil man im Fehlerfall, ja was dann?
Ich würde deinen Aufbau gerne nachbauen, bin aber noch am rätseln, ob wirklich dieses Projekt, oder doch lieber Venus OS auf dem Raspberry und den Hoymiles über NodeRed und das normale OpenDTU _selbst scripten_ besser für den Akku ist?

**Reply by berni2288** [reply]
Ja das läuft bei mir rund um die Uhr, die CAN-Schnittstelle wird nur zum Auslesen des SOC-Werts verwendet, für mehr nicht. Ich habe es auch monatelang ohne CAN-Kabel verwendet, also eine Kommunikation ist nicht unbedingt notwendig.
Das Pylontech BMS braucht so weit ich weiß keine Kommunikation weil es sich im Fehlerfall selbst abschalten kann. Also eine Tiefentladung oder eine Überladung sollte nicht möglich sein. Ich denke der Victron MPPT orientiert sich an der Batteriespannung und schaltet schon ab bevor die Batterie zu 100% geladen ist.
Ich reize allerdings auch die maximalen Lade/Entladeleistung der Pylontech-Batterie bei weitem nicht aus, da ich maximal mit 850 Watt lade und mit 800 Watt entlade.

Dieses Projekt ist im derzeitigen Status sicher nicht für Laien gedacht, es kann noch Bugs geben und man sollte unbedingt wissen wie es genau funktioniert und was man tut. Das Pylontech-BMS sollte aber Katastrophen verhindern.

**Comment by matthiasseemoo** [comment]
Wieso wird der Wechselrichter eigentlich direkt an der Batterie und nicht am Load Ausgang des Ladereglers angeschlossen?

**Reply by berni2288** [reply]
Der Victron Lastausgang kann laut Datenblatt bei 48V nur 1A liefern. Ich seh auch nicht wirklich einen Vorteil darin den Wechselrichter dort anzuschließen.

**Reply by Manos1966** [reply]
@berni2288 weiss du wo es im Datenblatt steht? Weil bei 48V nur 1A wurde 48Watt bedeuten. Und wenn ich das Datenblatt korrekt lese, er kann bei 48V bis zu 1160W von den PV Panels nehmen. Wenn er nur 48W davon liefern wuerde, das waere eine sehr schlechte Leistung.
![Capture](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/a09fba36-2424-4a04-bdae-644dd228a225)

**Reply by spcqike** [reply]
@Manos1966 der Ausgang ist in dem Fall nur ein steuersignal. Wenn man eine Batterie von den Lasten trennen will wird in dem Fall ein batteryprotect oder anderes Relais dazwischen geschalten. 

Im Falle von openDTU-onBattery braucht es gar nichts weiter. Man ließt einfach den Status auf dem Bus und aktiviert bzw. deaktiviert den wechselrichter. Dieser ist aber permanent direkt mit der Batterie verbunden. (Was auch empfohlen wird da beim trennen und wieder verbinden ja hohe Anlaufströme den wechselrichter beschädigen können, wenn keine zusätzlichen Maßnahmen ergriffen wurden) 

Nachtrag. Im Datenblatt steht das unter „laststrom 20/20/1“

**Reply by berni2288** [reply]
@Manos1966 Der Laderegler heißt ja "Laderegler" und soll bei mir nur die Batterie laden und nicht meinen Wechselrichter versorgen. Mein Wechselrichter hängt nicht am Laderegler-Lastausgang sondern direkt an der Batterie. Der Wechselrichter nimmt sich so viel Ampere Strom von der Batterie wie er grad benötigt. Der Lastausgang ist bei mir unbenutzt.

**Reply by Manos1966** [reply]
Danke @berni2288 & @spcqike ich habe es gefunden. Ich habe etwas ganz anderes bei "Lastausgang" verstanden.
![Kurzschlussbruecke](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/e20d55f1-66a0-428b-bcc3-c64c67aafc18)

**Comment by johannesosterberg** [comment]
Hallo, ich stehe gerade ein wenig auf dem Schlauch, muss ich das SN65HVD230 Board noch irgendwie an Ground vom CAN Port des Pylontech Akkus anschließen? Aus irgendeinem Grund bekomme ich keine Daten :)

**Reply by berni2288** [reply]
Hi,

schau mal in den Geräte/Pin manager ob die Pins dort richtig angezeigt werden.
Ansonsten würde ich empfehlen mal die Batterie neuzustarten und das Kabel bis zu den anschlusspins mit einem Multimeter oder so durchzuchecken und schau nochmal ob du eh die richtigen RJ45 pins verwendet hast. Die Pins 4 und 5.

**Reply by johannesosterberg** [reply]
Habe ich eigentlich alles überprüft, müssen die Pins irgendwelche speziellen sein, ich habe den kleineren D1 Mini und deshalb die Pins auf 33 und 27 gestellt. Was ich auch komisch finde ist, dass ich jetzt keine Konsolenwarnung mehr bekomme, dass der Pylontech Akku nicht gefunden werden kann. Habe erstmal alles Deaktiviert, außer den Canbus.

**Reply by berni2288** [reply]
Also am Gnd hab ich nichts angeschlossen, nur die 2 pins. Kenn ich mich mit dem Wemos D1 nicht aus, aber vermute dass die Pins eh passen.

**Comment by swingstate** [comment]
Es würde mich reizen das was Du auf der Prototypen Platinen gelötet hast in KiCad zu designen. Ob das sehr kompliziert ist? Eine Kleinserie wäre bestimmt drin bzw. würde auf Interesse stoßen.

**Reply by berni2288** [reply]
Wäre nett ja. Aber keine Ahnung, kenn mich damit nicht aus. Bin froh dass ich ein bisserl Löten kann :)

**Reply by iukbox** [reply]
Bin gerade dabei meine bestehende Platine zu erweitern weil das gewünscht wurde. Kannst nur leider selbst nicht testen da ich keinen Pylontech habe, sondern einen Eigenbau.
![Uploading IMG_20230625_201434.jpg…]()

**Comment by sbeneke** [comment]
Hallo @berni2288, ich habe grad dein Setup hier gefunden welches sich so ziemlich meinem ähnelt (ich nutze den HM-1500, Victron MPPT 100|20 und 2x Pylontech US200C). Mir sind nur so einige Werte nicht ganz klar wie ich sie am sinnvollsten einstelle. Magst du vielleicht die von dir eingestellten Werte vom Victron sowie die Dynamic Power limiter Settings zu deinem Setup zu posten? Ich glaube das könnte mir und vielen anderen gut helfen.

**Reply by berni2288** [reply]
Hey @sbeneke,

ja klar

Dynamic Power Limiter:
```
Aktiviert: ja
Aktiviere Solar Pass-trough: ja
Strategie zur Batterieentleerung: Leeren, wenn voll
Wechselrichter ID: ID 00
Kanal ID: CH 0
Erlaubter Stromverbrauch: 5W
Hysterese für den Zielstromverbrauch: 5W
Unteres Leistungslimit: 10W
Oberes Leistungslimit: 800W
Welchselrichter ist hinter Leistungsmesser: ja
Akku SOC - Start:25%
Akku SOC - Stop: 20%
DC Spannung - Start: 52V
DC Spannung - Stop:48V
DC Spannung - Lastkorrekturfaktor: 0,001V
```

Laderegler:
```
Batteriespannung: 48V
Batterievoreinstellung: Smart-Lithium (LiFePo4)
Lastausgang Betriebsmodus: Immer aus (verwend ich nicht)
Tx Port: Normale Kommunikation
Rx Port: Konfiguration Lastausgang (verwend ich aber nicht)
```

**Reply by sbeneke** [reply]
Top, vielen Dank, mit dem HM-600 funktioniert das auch so. Leider will das ganze mit dem HM-1500 nicht funktionieren bzw. der Mischbetrieb mit zwei Solarplatten direkt am Regler und zwei Platten über MPPT+Akku. Kann es sein dass du eine ältere Version hast? Bei mir gibt es noch drei weitere Werte zum Thema Solar Passthrough.

**Reply by berni2288** [reply]
Ja so ein Setup wurde wahrscheinlich nie getestet, wäre sicher besser das nicht zu mischen und wenn dann mit einem 2. Hoymiles. bei mir hängen alle Hoymiles-Anschlüsse an der gleichen Batterie. Ja meine Firmware ist sicher schon 2 Monate alt und hab grad keine Zeit für Updates. Bin froh dass alles funktioniert :)

**Reply by Manos1966** [reply]
Eigentlich, verstehe ich nicht, warum der HM-1500 nicht funktionieren wurde mit einem Setup 2 Baterie-Eingänge und 2 PV-Eingänge.

Ich habe es noch nicht probiert weil ich andere Baustellen habe, es steht aber auf meine Liste.

Die OpenDTU steuert die Wattleistung der HOYMILES.
Die Hoymiles selber steuert die vier Eingänge = zwei MPTTs.
So Lange du die Eingänge paarweise korrekt angeschlossen hast, sehe ich kein Grund warum es nicht funktionieren wurde.

**Comment by madmartin** [comment]
@berni2288 @sbeneke 

Noch ein Hinweis zur Einstellung des Victron MPPT zusammen mit Pylontech Batterien: da gibts ein explizites Papier von Victron

https://www.victronenergy.com/live/battery_compatibility:pylontech_phantom

In Abschnitt 5 stehen Parameter, die man im Laderegler einstellen soll:


<h4 id="charge_settings">5.2 Charge Settings</h4>
<div class="level4">

</div>

<h4 id="charger_tab">Charger tab</h4>
<div class="level4">
<div class="table sectionedit11">

Parameter | Setting
-- | --
Battery type | Lithium
Charge curve | Fixed
Absorption voltage | 52.0 V
Float voltage | 51.0 V
Absorption time | 1 Hr

</div>

<p>
Note: make sure to double check the float voltage after completing Assistants, and if necessary set it back to 51.0 V.
</p></div>




Die Spannungswerte sind per default  im Laderegler nach der Einstellung des Batterietyp LiFePo deutlich höher!

**Reply by sbeneke** [reply]
Sehr guter Hinweis! Ich habe mich schon über die recht hohen Werte gewundert. Habe es gleich im MPPT angepasst.

**Comment by alexz707** [comment]
Hallo @berni2288,
danke das du die Fotos mit uns teilst, ich würde gerne wissen, warum du mit der Sicherung "nur" den MPPT absicherst?
Müsste da nicht auch noch eine Sicherung zwischen Batterie und WR bzw. evtl auch zwischen MPPT und den Panels?

Was hast du denn mit der roten KFZ Sicherung abgesichert? (unten links im Bild?)

Danke & LG
Alex

**Reply by berni2288** [reply]
Hi,
diese Schmelzsicherung die du links unten siehst ist am Plus-Kabel zum WR. Also daher zwischen WR und Batterie.
Zwischen Victron und PV bin ich mir nicht sicher ob eine Sicherung da ist, wahrscheinlich nicht.
Falls du Verbesserungsvorschläge hast, bin ich offen dafür :)

**Comment by greymda** [comment]
Hi @berni2288  thanks for sharing your setup. I am building smth similar, can you share how many circuit breakers are you using and what specification? Cheers

**Comment by xmike13** [comment]
Wie sieht denn die Pinout.json für die waveshare Platine am opendtu on battery aus? (bei dir für Pin 14/27)
Nur das aktivieren der batterieschnittstelle und pylontech auswählen wird nicht reichen, oder?
